src/payment/strategies/affirm/affirm-payment-strategy.ts

```diff
@@ -126,17 +128,19 @@ export default class AffirmPaymentStrategy implements PaymentStrategy {
             shipping: this._getShippingAddress(),
             billing: this._getBillingAddress(),
             items: this._getItems(),
-            discounts: this._getDiscounts(),
             metadata: {
                 shipping_type: consignment.selectedShippingOption.type,
                 mode: 'modal',
+                platform_type: 'bigcommerce',
+                platform_version: '',
+                platform_affirm: '',
             },
+            discounts: this._getDiscounts(),
             order_id: order.orderId ? order.orderId.toString() : '',
-            shipping_amount: order.shippingCostTotal * 100,
-            tax_amount: order.taxTotal * 100,
-            total: order.orderAmount * 100,
+            shipping_amount: round(order.shippingCostTotal * 100, 0),
```

Comment:
Ideally, we shouldn't be converting dollars to cents like this. This is because the conversion is not always accurate as you're dealing with floats. Also, not all currencies have their main unit (i.e.: dollar) divided into 100 subunits (cents). Instead, we should ask the backend to return the original / integer amount directly. In here, we can simply forward the amount to Affirm. `orderAmountAsInteger` is an example of that approach.